libr/anal/p/anal_bf.c

```diff
@@ -51,12 +53,33 @@ static int bf_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int len) {
 						r_strbuf_setf (&op->esil,
 								"$$,brk,=[1],brk,++=,"
 								"ptr,[1],!,?{,0x%"PFMT64x",pc,=,brk,--=,}", dst);
-						break;
+						goto beach;
+					}
+				}
+				if (*p == 0x00 || *p == 0xff) {
+					op->type = R_ANAL_OP_TYPE_TRAP;
+					goto beach;
+				}
+				if (i == len - 1 && anal->esil->cb.resize_read_buf) {
+					const ut8 *new_buf;
+					new_buf = anal->esil->cb.resize_read_buf (anal, len + 1 + BUFSIZE_INC);
+					if (new_buf) {
+						if (buf_resized) {
```

Comment:
not a big fan of having boolean vars to check if a const* is in reality not const.

Reply:
`buf_resized` is there to differentiate between a buf allocated [on the stack](https://github.com/kazarmy/radare2/blob/6d03ad11f02a7e3b2263b0cb9385386631d35a90/libr/core/cmd_anal.c#L3108) and [on the heap](https://github.com/kazarmy/radare2/blob/6d03ad11f02a7e3b2263b0cb9385386631d35a90/libr/core/cmd_anal.c#L3094) ... doesn't seem to be any other way that is more elegant, unless you want to `strdup (buf)` at start which seems inefficient.

Reply:
nvm, removed `buf_resized` in 7881afc5c26afde4b465250568c40c21bd1ea470